lib/foreman/renderer.rb

```diff
@@ -54,7 +54,7 @@ def foreman_url(action = "built")
 
       url_for :only_path => false, :controller => "/unattended", :action => action,
               :protocol  => protocol, :host => host, :port => port, :script_name => path,
-              :token     => (@host.token.value unless @host.token.nil?)
+              :token     => (@host.token.value unless @host.try(:token).nil?)
```

Comment:
First of all, the local variable is not part of my commit and is not relevant to this pull request.

If you look at a couple of lines above this line, you'll see the line:
host     = uri.host

so, the local  'host' variable is the host from which the template can be downloaded. Whereas the instance variable "@host" will tell you which host has requested the template. At least, that is my understanding.

-PP